daemon/call_interfaces.c

```diff
@@ -680,6 +680,18 @@ static const char *call_offer_answer_ng(bencode_item_t *input, struct callmaster
 	if (sdp_streams(&parsed, &streams, &flags))
 		goto out;
 
+    /* Failover scenario because of timeout on offer response: siprouter tries
+     * to establish session with another rtpengine2 even though rtpengine1
+     * might have persisted part of the session. rtpengine2 deletes previous
+     * call in memory and recreates an OWN call in redis */
+	if (opmode == OP_OFFER) {
+		call = call_get(&callid, m);
+		if (call && IS_FOREIGN_CALL(call)) {
+			rwlock_unlock_w(&call->master_lock);
+			call_destroy(call);
+			obj_put(call);
+		}
```

Comment:
Hi Richard,

From the first glances and without testing I think I could manage to get the code into the call_get_or_create(). However, do you think that semantically the code snippet that deletes and then creates a call would fit right in that function? If so, I will make the changes there. test and update the pull request.

Alternatively I could change the current code to have obj_put(call) be executed if call is found, irrespective of IS_FOREIGN_CALL() result.

Which alternative would you prefer?

Thank you,
Lucian